subsys/ipc/ipc_service/backends/ipc_icmsg.c

```diff
@@ -187,21 +235,12 @@ static int send(const struct device *instance, void *token,
 
 const static struct ipc_service_backend backend_ops = {
 	.register_endpoint = register_ept,
+	.deregister_endpoint = deregister_ept,
 	.send = send,
 };
 
 static int backend_init(const struct device *instance)
```

Comment:
If this is empty function - just remove it. It does not make sense to keep it.

Reply:
I wish i could. If i remove it:
```
[00:00:00.251,617] <err> os: ***** USAGE FAULT *****
[00:00:00.251,647] <err> os:   Illegal use of the EPSR
[00:00:00.251,647] <err> os: r0/a1:  0x00009f98  r1/a2:  0xe000ed00  r2/a3:  0x20000a04
[00:00:00.251,678] <err> os: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x0000696d
[00:00:00.251,678] <err> os:  xpsr:  0x28000000
[00:00:00.251,708] <err> os: Faulting instruction address (r15/pc): 0x00000000
[00:00:00.251,739] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:00.251,770] <err> os: Current thread: 0x20000680 (unknown)
[00:00:00.308,197] <err> os: Halting system
```

I bet init kernel assumes initialization function exists for each device.

Reply:
Right. It seems like it tries to jump to NULL then.